test/bigtest/tests/suppress-from-discovery-tests.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { beforeEach, describe, it } from '@bigtest/mocha';
+import { expect } from 'chai';
+
+import { Form } from 'react-final-form';
+
+import { mountWithContext } from '../helpers/mountWithContext';
+
+import { SuppressFromDiscoveryFieldArray } from '../../../src/settings/components';
+import SuppressFromDiscoveryInteractor from '../interactors/suppress-from-discovery';
+
+const interactor = new SuppressFromDiscoveryInteractor();
+let formValues = {};
+
+describe('Suppress from discovery tests', () => {
+  describe('Rendering the "Suppress from discovery" settings component', () => {
+    beforeEach(async function () {
+      await mountWithContext(
+        <Form
+          onSubmit={() => {}}
+          render={({ handleSubmit, values }) => {
+            formValues = values;
+            return (
+              <form onSubmit={handleSubmit}>
+                <SuppressFromDiscoveryFieldArray name="test" />
+              </form>
+            );
+          }}
+          subscription={{ values: true }}
+        />
+      );
+    });
+
+    it('renders the description', () => {
+      expect(interactor.isDescription).to.be.true;
+    });
+
+    describe('agreement line checkbox tests', () => {
+      it('renders the agreement line checkbox', () => {
+        expect(interactor.isAgreementLine).to.be.true;
+      });
+
+      it('agreement line checkbox has the correct name prop', () => {
+        expect(interactor.agreementLineFieldName).to.equal('test.agreementLine');
```

Comment:
I think this test isn't necessary?

Reply:
I can remove it if you like, only added it because the way the HOC works is by matching the settingName to the fieldName, so it seemed important to call it out if that was changed